### PR TITLE
Add timestamp to Status "Fetch"

### PR DESCRIPTION
**Pull request by amitaibu**
#40

### CODE REVIEW

File: src/elm/Event.elm [hunk 1]
```diff
@@ -410,17 +411,16 @@ viewEventInfo model =
...
 getJson url accessToken =
   let
     encodedUrl = Http.url url [ ("access_token", accessToken) ]
   in
-    Http.get decodeData encodedUrl
-      |> Task.toResult
-      |> Task.map UpdateDataFromServer
-      |> Effects.task
-
+    Effects.task
```
Comment (amitaibu) [inline src/elm/Event.elm]: @rgrempel I was wondering if you could have a quick look here. I'm basically trying to get the timestamp after the `Http.get`, but I'm unable to get it working. Any ideas? I'm spending more time than I'm willing to admit to get a timestamp ;)

**Comment by rgrempel** [comment]
You were on the right track -- I took the liberty of pushing a fix (well, to make it compile, anyway).

* After adding a parameter to `Fetch` (i.e. the timestamp), you can't just test for `Fetched` via `==` any longer ... you have to do pattern matching, or use the little `isFetched` convenience function I added.

* This wasn't really a problem, but you had `UpdateDataFromServer` taking a Tuple2, which isn't really necessary -- it can just take two parameters without the tuple.

* As far as the `getJson` goes, you were basically on the right track, but it's hard to keep all of the manipulation straight -- especially the precedence. So, what I often do is use `let` to give things names, and use explicit parentheses with the lambdas -- otherwise, I can't keep track of what the precedence actually is.

**Comment by amitaibu** [comment]
@rgrempel thank you! this is really great :)

As `getCurrentTime` is a task, wouldn't we need to do `andThen` after it as a good practice (eventhough we know its a sync task)

**Comment by rgrempel** [comment]
Task.map is also async -- that is, it also only gets applied after the Task actually completes. The only reason to use Task.andThen would be if you need to chain yet another Task. Which isn't needed here -- all you have to do is construct the Action, which is sync (at that point -- since you now have all the ingredients).

**Comment by amitaibu** [comment]
> Task.map is also async

:+1:

**Comment by amitaibu** [comment]
Merged, thanks!

**Pull request closed without merging**